sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -157,13 +157,12 @@ class BulkWriter(container: CosmosAsyncContainer,
                 captureIfFirstFailure(resp.getException)
                 cancelWork()
             }
-          } else if (!itemResponse.isSuccessStatusCode) {
+          } else if (Option(itemResponse).isEmpty || !itemResponse.isSuccessStatusCode) {
```

Comment:
Why are we adding this new check?
If itemResponse can be empty, then we should also handle it appropriately when passing it through `handleNonSuccessfulStatusCode` API.

Reply:
Done below - motivation for the change were transient unit test failures. I debugged those and saw that on 500 - Internal Server error response is null (and NPE happened here). While debugging the unit tests AEMO also reported same NPE in a support ticket.

Reply:
Makes sense!